src/MessageList/MessageListItem.vala

```diff
@@ -388,13 +388,45 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
                 return;
             }
 
-            message_content = (string) os.steal_data ();
+            // Convert the message to UTF-8 to ensure we have a valid GLib string.
+            message_content = convert_to_utf8(os, field.param ("charset"));
```

Comment:
Missing a space between function name and parenthesis opening

Reply:
Ouch, forgot to check for code style in today's commits. I should probably write a Builder plugin :)